xarray/plot/plot.py

```diff
@@ -941,3 +1013,82 @@ def pcolormesh(x, y, z, ax, infer_intervals=None, **kwargs):
         ax.set_ylim(y[0], y[-1])
 
     return primitive
+
+
+def scatter(ds, x, y, hue=None, col=None, row=None,
+            col_wrap=None, sharex=True, sharey=True, aspect=None,
+            size=None, subplot_kws=None, add_legend=None,
+            discrete_legend=None, **kwargs):
+
+    if kwargs.get('_meta_data', None):
+        discrete_legend = kwargs['_meta_data']['discrete_legend']
+    else:
+        meta_data = _infer_scatter_meta_data(ds, x, y, hue,
+                                             add_legend, discrete_legend)
+        discrete_legend = meta_data['discrete_legend']
+        add_legend = meta_data['add_legend']
+
+    if col or row:
+        ax = kwargs.pop('ax', None)
+        figsize = kwargs.pop('figsize', None)
+        if ax is not None:
+            raise ValueError("Can't use axes when making faceted plots.")
+        if aspect is None:
+            aspect = 1
+        if size is None:
+            size = 3
+        elif figsize is not None:
+            raise ValueError('cannot provide both `figsize` and '
+                             '`size` arguments')
+
+        g = FacetGrid(data=ds, col=col, row=row, col_wrap=col_wrap,
+                      sharex=sharex, sharey=sharey, figsize=figsize,
+                      aspect=aspect, size=size, subplot_kws=subplot_kws)
+        return g.map_scatter(x=x, y=y, hue=hue, add_legend=add_legend,
+                             discrete_legend=discrete_legend, **kwargs)
+
+    data = _infer_scatter_data(ds, x, y, hue, discrete_legend)
+
+    figsize = kwargs.pop('figsize', None)
+    ax = kwargs.pop('ax', None)
+    ax = get_axis(figsize, size, aspect, ax)
+    if discrete_legend:
+        primitive = ax.plot(data['x'], data['y'], '.')
```

Comment:
You should be able to avoid switching between `plot()` and `scatter()` by building a color array, using `groupby(color)` to loop over the color groups and using `scatter()` on each group. Then make a legend. (I'm following https://matplotlib.org/gallery/lines_bars_and_markers/scatter_with_legend.html)